### PR TITLE
Prevent ssh config tests from corrupting local ssh config file

### PR DESCRIPTION
**Pull request by alanhamlett**
Instead of overwriting local `~/.ssh/config` file with the testdata file, we mock out `ssh_config` to point it to a temp file.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -53,3 +53,5 @@ require (
 replace github.com/alecthomas/chroma => github.com/wakatime/chroma v0.10.0-wakatime.1
 
 replace github.com/matishsiao/goInfo => github.com/wakatime/goInfo v0.1.0-wakatime.8
+
+replace github.com/kevinburke/ssh_config => github.com/wakatime/ssh_config v1.2.0-wakatime.1
```
Comment (gandarez) [inline go.mod]: Why do we need to replace this library?

File: pkg/remote/remote_test.go [hunk 2]
```diff
@@ -106,22 +105,20 @@ func TestWithDetection_sshConfig_Hostname(t *testing.T) {
 	shutdown, host, port := testServer(t, false)
 	defer shutdown()
 
-	sshConfigFile, _ := homedir.Expand("~/.ssh/config")
+	tmpFile, err := os.CreateTemp(t.TempDir(), "")
+	require.NoError(t, err)
+
+	defer tmpFile.Close()
+
+	sshConfigFile := tmpFile.Name()
```
Comment (gandarez) [inline pkg/remote/remote_test.go]: It's being used once in the func so don't need to create a new variable for that, simply use `tmpFile.Name()` instead.

File: pkg/remote/remote_test.go [hunk 3]
```diff
@@ -184,22 +181,20 @@ func TestWithDetection_sshConfig_UserKnownHostsFile_mismatch(t *testing.T) {
 	shutdown, host, port := testServer(t, true)
 	defer shutdown()
 
-	sshConfigFile, _ := homedir.Expand("~/.ssh/config")
+	tmpFile, err := os.CreateTemp(t.TempDir(), "")
+	require.NoError(t, err)
+
+	defer tmpFile.Close()
+
+	sshConfigFile := tmpFile.Name()
```
Comment (gandarez) [inline pkg/remote/remote_test.go]: same here

File: pkg/remote/remote_test.go [hunk 4]
```diff
@@ -256,22 +251,20 @@ func TestWithDetection_sshConfig_UserKnownHostsFile_match(t *testing.T) {
...
```
Comment (gandarez) [inline pkg/remote/remote_test.go]: same here

**Review by gandarez — CHANGES_REQUESTED** [review]
IMO we don't need to replace `ssh_config` if there isn't any bug fix or enhancement to the forked repo.

**Comment by alanhamlett** [comment]
> IMO we don't need to replace `ssh_config` if there isn't any bug fix or enhancement to the forked repo.

We do need the latest change that makes `ConfigFinder` public so we can overwrite it with out mocked one.

File: go.mod (hunk 1 above)
Comment (alanhamlett) [reply go.mod]: We need this new public function recently added so we can mock the ssh config file location:
https://github.com/kevinburke/ssh_config/commit/5724bd127982289dd52e79933bbe60b98a5b639e#diff-0e426a43248661127a0c0ee115aef7a1093b635f8993b3f7ebb1dd9f05b8f249R267

Comment (alanhamlett) [reply go.mod]: Updated to use master instead of fork.

**Review by gandarez — APPROVED** [review]
Looks good!

**Pull request merged**